lib/services/dashd.js

```diff
@@ -2248,16 +2248,8 @@ Dash.prototype.getDetailedTransaction = function(txid, callback) {
     }
   }
 
-  function addTxlockToTx(tx,result) {
-    tx.txlock = false;
-    if (tx.feeSatoshis >= Dash.DEFAULT_INSTANTSEND_FEE) { // if transaction fee is >= 0.0001 check for txlock
-      var rawTx = new dashcore.Transaction(result.hex);
-      var hash = dashcore.crypto.Hash.sha256sha256(rawTx.toBuffer());
-      var id = hash.toString('binary');
-      if (self.node.services.dashd.zmqKnownTransactionLocks.get(id)) {
-        tx.txlock = true; // if "txlock" notification received through ZMQ, set txlock = true
-      }
-    }
+  function addTxlockToTx(tx, result) {
+    tx.txlock = result.instantlock || result.chainlock;
```

Comment:
But chainlock isn't the same thing as IS lock, isn't it? IS lock prevents double spend within one block, chainlock just blocks possibility of the reorgs after the block was mined

Reply:
I think from a user perspective it's kinda the same - a transaction is confirmed.

Reply:
@antouhou please check the relevant exchange in 14-0-dev where this topic was topic was discussed. Esentially with a chainlock the instantlock field is set to true as well. So there will never be a situation where `.chainlock` is true and `.instantlock` is false.
https://dashpay.slack.com/archives/G7W56UMCG/p1553618255080300?thread_ts=1553578894.065300&cid=G7W56UMCG